### PR TITLE
external auth providers & k8s [ch8919]

### PR DESCRIPTION
**Pull request by ericpaulsen**
[related clubhouse story ch8919](https://app.clubhouse.io/coder/story/8919/guide-for-how-to-configure-a-kubernetes-cluster-to-use-external-auth-provider-for-rbac).

this is meant as an MVP before we flesh out a full guide as to how to configure a cluster to use external auth providers in tandem with RBAC.

### CODE REVIEW

File: setup/kubernetes/google.md [hunk 1]
```diff
@@ -78,6 +78,18 @@ gcloud beta container --project "$PROJECT_ID" \
...
 
+## Access Control
+
+GKE allows for integration between Identity Access and Management (IAM) and Kubernetes'
+native Role-Based Access Control (RBAC) mechanism to authorize user actions in the
+cluster. IAM configuration is primarily applied at the Google Cloud Project level,
+and all clusters within that Project. Kubernetes RBAC configuration applies to each
+individual cluster, allowing for fine-grained authorization down to the namespace
+level.
+
```
Comment (ericpaulsen) [inline setup/kubernetes/google.md:89]: future state - we could reference the diagrams [in this article](https://medium.com/uptime-99/making-sense-of-kubernetes-rbac-and-iam-roles-on-gke-914131b01922#CommonIAMandRBACRoles) to clarify the differences between GKE & K8s roles.

File: setup/kubernetes/google.md [hunk 2]
```diff
@@ -78,6 +78,18 @@ gcloud beta container --project "$PROJECT_ID" \
...
+For more information, see [Using Kubernetes RBAC auhtorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
```
Comment (jawnsy) [inline setup/kubernetes/google.md:90]: FYI, drive-by commenter :car: auhtorization -> authorization

If you're using VSCode to edit, I've found the [Code Spell Checker extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) to be excellent... I should add that to our recommendations in m :thinking:

Comment (ericpaulsen) [reply setup/kubernetes/google.md:90]: ah good catch - thank you sir

**Review by khorne3 — APPROVED** [review]
